python/cudf/cudf/core/column/categorical.py

```diff
@@ -344,6 +344,10 @@ def element_indexing(self, index):
         val = self.as_numerical.element_indexing(index)
         return self._decode(val) if val is not None else val
 
+    @property
+    def __cuda_array_interface__(self):
+        return self.cat().codes.__cuda_array_interface__
```

Comment:
This feels like unexpected behavior to me. Should we throw an exception here instead that says `__cuda_array_interface__` isn't supported for Categorical typed columns, if you want the codes, use `.codes` accessor.

Reply:
Thanks @kkraus14. This was @shwina's suggestion [above]( https://github.com/rapidsai/cudf/pull/3567#pullrequestreview-330603325 ). Don't have strong feelings about what we do here personally. Happy to defer. Please let me know if/how this should be updated 🙂

Reply:
We should throw an exception here and guide a user to using `.codes` or `.categories` instead.

Reply:
Sounds good to me!

Reply:
Ok added an exception. Please take a look and let me know what you think 🙂